include/vcpkg/base/strings.h

```diff
@@ -47,7 +47,7 @@ namespace vcpkg::Strings::details
 
 namespace vcpkg::Strings
 {
-#ifdef __APPLE__
+#ifndef __cpp_lib_boyer_moore_searcher
```

Comment:
I think it's better to flip this condition. The inclusion of the experimental header should also be adjusted.